Dockerfile.aro-azext

```diff
@@ -0,0 +1,7 @@
+FROM mcr.microsoft.com/azure-cli:2.58.0
+
+RUN pip install pytest
+COPY /python /data/
+
+WORKDIR /data/az/aro
+CMD ["/bin/sh", "-c", "pytest --ignore=azext_aro/tests/latest/integration; python3 setup.py bdist_wheel"]
```

Comment:
I would prefer if the `podman build` itself got us our extension built and ready to go, rather than requiring both `build` and `run` commands to generate an artifact. Can we:

1. do a `COPY` for all the files required to build (now, that's a `-v` in the makefile)
2. Make this `CMD` a `RUN` clause instead

This means our makefile becomes a single `podman build`, and by the end, we have a guaranteed known-good artifact (docker image) with all our changes. If we want those changes locally, we can issue a `podman cp` to extract it, but for now I'd like to just keep things 100% in the container space (and prevent artifact dumping in our git).